trucks.js

```diff
@@ -501,4 +501,196 @@ var foodTrucks = [
 		Facebook: 'https://www.facebook.com/fulltilticecreams',
 		Twitter: 'https://twitter.com/fticecream'
 	}
-];
\ No newline at end of file
+];
+
+/*
+ * Homework 
+ * JSCRIPT300-Spring2015/Module_4b
+ * by Diane Zevenbergen
+ *
+ * NOTE: I revised code from HW for Module 4a in order to use
+ * underscore in getTruck, getFoodTypes and filterByFoodType
+ */
+
+'use strict';
+
+var _ = require('underscore');
+
+/* 
+ * Function returns array object of food trucks available today.
+ * Uses underscore's filter() method.
+ */ 
+function filterByDay(day) {
+
+    // Filter returns the subset of foodTrucks that match 'day'
+    var todaysTrucks = _.filter(foodTrucks, function(item) {
+
+        // Exclude truck if object doesn't include 'schedule' prop, otherwise check for 'day' match
+        if (typeof item.schedule === 'undefined') {
+            return false;
+        } else {
+            return (item.schedule.indexOf(day) !== -1);
+        }        
+    });
+    
+    return todaysTrucks;
+}
+
+// Function returns all truck objects
+function getTrucks() {
+    return foodTrucks;
+}
+
+// Function returns object info for matching truck if found
+function getTruck(name) {
+    var matchingTruck = {};
+    
+    name = name.toLowerCase().trim();
+
+    // Use underscore to find matching truck
+    matchingTruck = _.find(foodTrucks, function (i) { 
+        if (i.name.toLowerCase().trim() === name) {
+            return i;
+        }
+    });
+    
+    // Return matchingTruck
+    return matchingTruck;
+}
+
+//Function returns array of unique food types
+function getFoodTypes() {
+    var uniqueFoods = [];
+    var index = 0;
+
+    // Returns list of unique food types for all trucks
+	uniqueFoods = _.uniq(_.flatten(_.pluck(foodTrucks, "type")));
+    
+    // Sorts list, case-insensitive (from stackoverflow)
+    uniqueFoods = _.sortBy(uniqueFoods, function (i) { return i.toLowerCase(); });
+
+    // Return array of unique food types
+    return uniqueFoods;
+  
+}
+
+// Function returns object info for matching food type (if found)
+function filterByFoodType(type) {
+    var matchingTrucks= [];
+    
+    type = type.toLowerCase();
+
+    /*
+     * Filter returns the trucks that sell a specific food type 'type'
+     * QUESTION:  I added a prototype method make it case insensitive -
+     * is there a better way?
+     */
+	var matchingTrucks = _.filter(foodTrucks, function (item) {
+                
+		if (typeof item.type === 'undefined') {
+			return false;
+		} else {
+            return (item.type.toLowerCase().indexOf(type) !== -1);
+		}
+	});
+
+    return matchingTrucks;    
+}
+
+
+// Prototype method returns a lowercase version of the array that calls it (found concept on Stackoverflow, changed somewhat)
+Array.prototype.toLowerCase = function() { 
```

Comment:
I would avoid modifying built-in prototypes. This will require a loop through each type array on a truck object anyway, so better to just loop through each value and call the build-in toLowerCase() method on each string before comparing.